frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,17 +1,40 @@
+import * as _ from 'lodash-es';
 import * as React from 'react';
 import * as PropTypes from 'prop-types';
-import { Heading } from './utils';
+import { Heading, CopyToClipboard } from './utils';
 
-export const ConfigMapData = ({data}) => {
+export const MaskedData = () => {
+  return <React.Fragment>
+    <span className="sr-only">value hidden</span>
+    <span aria-hidden="true">&bull;&bull;&bull;&bull;&bull;</span>
+  </React.Fragment>;
+};
+
+export const KeyValueData = ({data, showSecret}) => {
```

Comment:
I have to disagree with @benjaminapetersen here... I think this was cleaner without trying to share a component for the data. They're as different as they are the same.

Reply:
@benjaminapetersen not quite sure if collapsing `ConfigMapData` and `SecretData` would do any good, since they are fairly different. For `SecretData` we need to keep the state of the `showSecret` and also we need the `toggleSecret` function that is bind to `this`. For that reason it make sense to have it as class component
On the other hand `ConfigMapData` just displays data, so making it a functional component is the right approach imho.